nebula_ros/include/nebula_ros/common/single_consumer_processor.hpp

```diff
@@ -0,0 +1,157 @@
+// Copyright 2025 TIER IV, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <condition_variable>
+#include <functional>
+#include <mutex>
+#include <queue>
+#include <thread>
+#include <utility>
+
+namespace nebula::ros
+{
+
+/// @brief A thread-safe single-consumer queue processor that runs a callback function
+/// on items in a separate thread. The queue has a maximum size, so insertions block or
+/// are rejected if the consumer is slower than the producer.
+/// @tparam T The type of items to process
+template <typename T>
+class SingleConsumerProcessor
+{
+public:
+  using callback_t = std::function<void(T &&)>;
+
+  /// @brief Constructor
+  /// @param callback The callback function to execute on each item
+  /// @param max_queue_size The maximum size of the queue
+  explicit SingleConsumerProcessor(callback_t callback, size_t max_queue_size)
+  : callback_(std::move(callback)), max_queue_size_(max_queue_size)
+  {
+    if (max_queue_size == 0) {
+      throw std::invalid_argument("Max queue size must be greater than 0");
+    }
+
+    if (!callback_) {
+      throw std::invalid_argument("Callback function must not be null");
+    }
+
+    consumer_thread_ = std::thread(&SingleConsumerProcessor::consumer_loop, this);
+  }
+
+  /// @brief Destructor - stops the processing thread
+  ~SingleConsumerProcessor()
+  {
+    stop();
+    consumer_thread_.join();
```

Comment:
`stop()` also contains `consumer_thread_.join();`, so should this line be removed to prevent a double `join()` ?